sycl/include/CL/sycl/detail/common.hpp

```diff
@@ -83,7 +83,7 @@ static inline std::string codeToString(cl_int code){
 #endif
 
 #if __has_attribute(always_inline)
-#define ALWAYS_INLINE __attribute__((always_inline))
+#define ALWAYS_INLINE __attribute__((always_inline)) inline
```

Comment:
How this modification related to the adding check for nan values?

Reply:
I use sycl builtins to check for nan/inf values. Usage of these builtins causes warnings. Purpose of this change is to fix these warnings.

Reply:
You are changing the define semantics w/o updating the name!
Can you just apply `inline` keyword to specific built-ins you are using?
Also I'm not sure I follow what warning is emitted, so it's hard to say what is the right fix.

Reply:
Moved inline keyword from macro to builtin declaration.

Short description of the problem:
According to always_inline attribute description in gcc's documentation https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html:
`For functions *declared inline*, this attribute inlines the function independent of any restrictions that otherwise apply to inlining. `
This means that if declaration is not marked inline then warning will be generated.

Problem could be reproduced in the following test case:
`#include <CL/sycl.hpp>

int main() {
  return 0;
}`

When compiling with gcc 5.4.0:
g++ -std=c++11  -I$PATH_TO_SYCL/sycl/include test.cpp -c
gives the bunch of warnings:
"sycl/include/CL/sycl/detail/builtins.hpp:32:19: warning: always_inline function might not be inlinable [-Wattributes]"


If you think that I shouldn't fix this in this patch or my fix doesn't look good then I can create issue and assign it to the author.